core/src/test/java/org/infinispan/functional/FunctionalL1Test.java

```diff
@@ -0,0 +1,68 @@
+package org.infinispan.functional;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+
+import java.io.Serializable;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.api.functional.EntryView.ReadEntryView;
+import org.infinispan.commons.api.functional.EntryView.WriteEntryView;
+import org.infinispan.commons.api.functional.FunctionalMap;
+import org.infinispan.commons.marshall.MarshallableFunctions;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.functional.impl.FunctionalMapImpl;
+import org.infinispan.functional.impl.ReadWriteMapImpl;
+import org.infinispan.functional.impl.WriteOnlyMapImpl;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "functional.FunctionalL1Test")
+public class FunctionalL1Test extends AbstractFunctionalOpTest {
+   @Override
+   protected void configureCache(ConfigurationBuilder builder) {
+      super.configureCache(builder);
+      if (builder.clustering().cacheMode().isDistributed()) {
+         builder.clustering().l1().enable();
+      }
+   }
+
+   @Test(dataProvider = "owningModeAndWriteMethod")
+   public void testEntryInvalidated(boolean isOwner, WriteMethod method) {
+      Cache<Object, String> primary = cache(isOwner ? 0 : 2, DIST);
+      Cache<Object, String> backup = cache(1, DIST);
+      Cache<Object, String> reader = cache(3, DIST);
+      Cache<Object, String> nonOwner = cache(isOwner ? 2 : 0, DIST);
+      Object KEY = getKeyForCache(primary, backup);
+      primary.put(KEY, "value");
+
+      assertEquals(null, reader.getAdvancedCache().getDataContainer().get(KEY));
```

Comment:
I would, but regrettably AssertJUnit then says only 'value was expected to be null' and does not specify that; It's more useful for debugging to use `assertEquals(null, ...)`